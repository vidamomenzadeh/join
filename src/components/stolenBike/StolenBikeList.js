import StolenBikeItem from '@components/stolenBike/StolenBikeItem';
import React from 'react';
import {Pagination} from "antd";
import PropTypes from 'prop-types';

export default function StolenBikeList(props){
    const {bikes, fetchBikesOnPage, total, currentPage} = props;

    return (
        <div className="bikes-list-wrapper">
            {Object.keys(bikes).map(function(id) {
                return <StolenBikeItem
                    id={id}
                    key={id}
                    bike={bikes[id]}/>;
            })}
            <Pagination defaultCurrent={currentPage} total={total} onChange={(page)=>{
                fetchBikesOnPage(page);
            }}/>
        </div>
    );
}

StolenBikeList.propTypes = {
    bikes: PropTypes.object,
}
