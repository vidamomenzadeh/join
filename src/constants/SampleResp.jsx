export const SAMPLE_RESP = {"1" :{
    "id": "1",
    "title": "Stolen Bike",
    "description": "Stolen Bike des.",
    "address": "Berlin",
    "occurred_at": 1554543180609,
    "updated_at": 1554543180609,
    "source": {
        "name": null,
        "html_url": null,
        "api_url": ""
    },
    "media": {
        "image_url": null,
        "image_url_thumb": null
    },
    "location_type": null,
    "location_description": null,
    "type": "Theft",
    "type_properties": null
}};

export const SAMPLE_RESP_2 = {"incidents":[{
        "id": "1",
        "title": "Stolen Bike",
        "description": "Stolen Bike des.",
        "address": "Berlin",
        "occurred_at": 1554543180609,
        "updated_at": 1554543180609,
        "source": {
            "name": null,
            "html_url": null,
            "api_url": ""
        },
        "media": {
            "image_url": null,
            "image_url_thumb": null
        },
        "location_type": null,
        "location_description": null,
        "type": "Theft",
        "type_properties": null
    }]};
