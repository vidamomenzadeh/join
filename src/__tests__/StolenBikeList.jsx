import React from "react";
import { shallow, mount } from 'enzyme';
import StolenBikeList from '@components/stolenBike/StolenBikeList';

import enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {SAMPLE_RESP} from "@constants/SampleResp";
enzyme.configure({ adapter: new Adapter() });

const react = require("react");
module.exports = { ...react, memo: x => x };

describe('StolenBikeList Render correctly', () => {

    const bikes = SAMPLE_RESP;

    it('StolenBikeList snapshot test Without bikes', () => {
        const stolenBikeList = shallow(<StolenBikeList bikes={{}} />);
        expect(stolenBikeList).toMatchSnapshot();
    });

    it('StolenBikeList should render correctly with bikes', () => {
        const stolenBikeList = mount(<StolenBikeList bikes={bikes} />);
        expect(stolenBikeList).toMatchSnapshot();
        expect(stolenBikeList.find('.bikes-list-wrapper')).toHaveLength(1);
    });

});

