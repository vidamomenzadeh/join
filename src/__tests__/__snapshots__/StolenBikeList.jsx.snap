// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StolenBikeList Render correctly StolenBikeList should render correctly with bikes 1`] = `
<StolenBikeList
  bikes={
    Object {
      "1": Object {
        "address": "Berlin",
        "description": "Stolen Bike des.",
        "id": "1",
        "location_description": null,
        "location_type": null,
        "media": Object {
          "image_url": null,
          "image_url_thumb": null,
        },
        "occurred_at": 1554543180609,
        "source": Object {
          "api_url": "",
          "html_url": null,
          "name": null,
        },
        "title": "Stolen Bike",
        "type": "Theft",
        "type_properties": null,
        "updated_at": 1554543180609,
      },
    }
  }
>
  <div
    className="bikes-list-wrapper"
  >
    <StolenBikeItem
      bike={
        Object {
          "address": "Berlin",
          "description": "Stolen Bike des.",
          "id": "1",
          "location_description": null,
          "location_type": null,
          "media": Object {
            "image_url": null,
            "image_url_thumb": null,
          },
          "occurred_at": 1554543180609,
          "source": Object {
            "api_url": "",
            "html_url": null,
            "name": null,
          },
          "title": "Stolen Bike",
          "type": "Theft",
          "type_properties": null,
          "updated_at": 1554543180609,
        }
      }
      id="1"
      key="1"
    >
      <div
        className="bike-item"
        id="bike-item-1"
      >
        <img
          className="noImage"
          src="/public/img/bike.svg"
        />
        <div
          className="bike-content"
        >
          <div
            className="bike-title"
          >
            Stolen Bike
          </div>
          <div>
            <label>
              Description:
            </label>
            Stolen Bike des.
          </div>
          <div>
            <label>
              Location:
            </label>
            Berlin
          </div>
          <div>
            <label>
              Date of the theft:
            </label>
            6 April 2019
          </div>
          <div>
            <label>
              Date of the Report:
            </label>
            6 April 2019
          </div>
        </div>
      </div>
    </StolenBikeItem>
    <Pagination
      onChange={[Function]}
    >
      <LocaleReceiver
        componentName="Pagination"
        defaultLocale={
          Object {
            "items_per_page": "/ page",
            "jump_to": "Goto",
            "jump_to_confirm": "confirm",
            "next_3": "Next 3 Pages",
            "next_5": "Next 5 Pages",
            "next_page": "Next Page",
            "page": "",
            "prev_3": "Previous 3 Pages",
            "prev_5": "Previous 5 Pages",
            "prev_page": "Previous Page",
          }
        }
      >
        <Pagination
          className=""
          defaultCurrent={1}
          defaultPageSize={10}
          hideOnSinglePage={false}
          itemRender={[Function]}
          jumpNextIcon={
            <a
              className="ant-pagination-item-link"
            >
              <div
                className="ant-pagination-item-container"
              >
                <Icon
                  className="ant-pagination-item-link-icon"
                  type="double-right"
                />
                <span
                  className="ant-pagination-item-ellipsis"
                >
                  •••
                </span>
              </div>
            </a>
          }
          jumpPrevIcon={
            <a
              className="ant-pagination-item-link"
            >
              <div
                className="ant-pagination-item-container"
              >
                <Icon
                  className="ant-pagination-item-link-icon"
                  type="double-left"
                />
                <span
                  className="ant-pagination-item-ellipsis"
                >
                  •••
                </span>
              </div>
            </a>
          }
          locale={
            Object {
              "items_per_page": "/ page",
              "jump_to": "Goto",
              "jump_to_confirm": "confirm",
              "next_3": "Next 3 Pages",
              "next_5": "Next 5 Pages",
              "next_page": "Next Page",
              "page": "",
              "prev_3": "Previous 3 Pages",
              "prev_5": "Previous 5 Pages",
              "prev_page": "Previous Page",
            }
          }
          nextIcon={
            <a
              className="ant-pagination-item-link"
            >
              <Icon
                type="right"
              />
            </a>
          }
          onChange={[Function]}
          onShowSizeChange={[Function]}
          prefixCls="ant-pagination"
          prevIcon={
            <a
              className="ant-pagination-item-link"
            >
              <Icon
                type="left"
              />
            </a>
          }
          selectComponentClass={[Function]}
          selectPrefixCls="ant-select"
          showLessItems={false}
          showPrevNextJumpers={true}
          showQuickJumper={false}
          showSizeChanger={false}
          showTitle={true}
          style={Object {}}
          total={0}
        >
          <ul
            className="ant-pagination "
            style={Object {}}
            unselectable="unselectable"
          >
            <li
              aria-disabled={true}
              className="ant-pagination-disabled ant-pagination-prev"
              onClick={[Function]}
              onKeyPress={[Function]}
              tabIndex={null}
              title="Previous Page"
            >
              <a
                className="ant-pagination-item-link"
              >
                <Icon
                  type="left"
                >
                  <LocaleReceiver
                    componentName="Icon"
                  >
                    <i
                      aria-label="icon: left"
                      className="anticon anticon-left"
                    >
                      <IconReact
                        className=""
                        type="left-o"
                      >
                        <svg
                          aria-hidden="true"
                          className=""
                          data-icon="left"
                          fill="currentColor"
                          height="1em"
                          key="svg-left"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                            key="svg-left-svg-0"
                          />
                        </svg>
                      </IconReact>
                    </i>
                  </LocaleReceiver>
                </Icon>
              </a>
            </li>
            <Pager
              className="ant-pagination-disabled"
              itemRender={[Function]}
              key="noPager"
              locale={
                Object {
                  "items_per_page": "/ page",
                  "jump_to": "Goto",
                  "jump_to_confirm": "confirm",
                  "next_3": "Next 3 Pages",
                  "next_5": "Next 5 Pages",
                  "next_page": "Next Page",
                  "page": "",
                  "prev_3": "Previous 3 Pages",
                  "prev_5": "Previous 5 Pages",
                  "prev_page": "Previous Page",
                }
              }
              onClick={[Function]}
              onKeyPress={[Function]}
              page={0}
              rootPrefixCls="ant-pagination"
              showTitle={true}
            >
              <li
                className="ant-pagination-item ant-pagination-item-0 ant-pagination-disabled ant-pagination-item-disabled"
                onClick={[Function]}
                onKeyPress={[Function]}
                tabIndex="0"
                title={0}
              >
                <a>
                  0
                </a>
              </li>
            </Pager>
            <li
              aria-disabled={true}
              className="ant-pagination-disabled ant-pagination-next"
              onClick={[Function]}
              onKeyPress={[Function]}
              tabIndex={null}
              title="Next Page"
            >
              <a
                className="ant-pagination-item-link"
              >
                <Icon
                  type="right"
                >
                  <LocaleReceiver
                    componentName="Icon"
                  >
                    <i
                      aria-label="icon: right"
                      className="anticon anticon-right"
                    >
                      <IconReact
                        className=""
                        type="right-o"
                      >
                        <svg
                          aria-hidden="true"
                          className=""
                          data-icon="right"
                          fill="currentColor"
                          height="1em"
                          key="svg-right"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
                            key="svg-right-svg-0"
                          />
                        </svg>
                      </IconReact>
                    </i>
                  </LocaleReceiver>
                </Icon>
              </a>
            </li>
            <Options
              changeSize={null}
              current={1}
              goButton={false}
              locale={
                Object {
                  "items_per_page": "/ page",
                  "jump_to": "Goto",
                  "jump_to_confirm": "confirm",
                  "next_3": "Next 3 Pages",
                  "next_5": "Next 5 Pages",
                  "next_page": "Next Page",
                  "page": "",
                  "prev_3": "Previous 3 Pages",
                  "prev_5": "Previous 5 Pages",
                  "prev_page": "Previous Page",
                }
              }
              pageSize={10}
              pageSizeOptions={
                Array [
                  "10",
                  "20",
                  "30",
                  "40",
                ]
              }
              quickGo={null}
              rootPrefixCls="ant-pagination"
              selectComponentClass={[Function]}
              selectPrefixCls="ant-select"
            />
          </ul>
        </Pagination>
      </LocaleReceiver>
    </Pagination>
  </div>
</StolenBikeList>
`;

exports[`StolenBikeList Render correctly StolenBikeList snapshot test Without bikes 1`] = `
<div
  className="bikes-list-wrapper"
>
  <Pagination
    onChange={[Function]}
  />
</div>
`;
